core/lib/src/fs/maybe_zipped_file.rs

```diff
@@ -0,0 +1,110 @@
+use std::ffi::OsString;
+use std::path::Path;
+use std::io;
+
+use rocket_http::ContentType;
+
+use crate::fs::NamedFile;
+use crate::response::{self, Responder};
+use crate::Request;
+
+/// A [`Responder`] that looks for pre-zipped files on the filesystem (by an 
+/// extra `.gz' file extension) to serve in place of the given file.
+///
+/// # Example
+///
+/// A simple static file server mimicking [`FileServer`]:
+///
+/// ```rust
+/// # use rocket::get;
+/// use std::path::{PathBuf, Path};
+///
+/// use rocket::fs::{NamedFile, relative};
+///
+/// #[get("/file/<path..>")]
+/// pub async fn second(path: PathBuf) -> Option<NamedFile> {
+///     let mut path = Path::new(relative!("static")).join(path);
+///     if path.is_dir() {
+///         path.push("index.html");
+///     }
+///
+///     MaybeZippedFile::open(path).await.ok()
+/// }
+/// ```
+///
+/// Always prefer to use [`FileServer`] which has more functionality and a
+/// pithier API.
+///
+/// [`FileServer`]: crate::fs::FileServer
+#[derive(Debug)]
+pub(crate) struct MaybeZippedFile {
```

Comment:
It may be a good idea to change `MaybeZippedFile` into something more generic like `MaybeCompressedFile`. Reason being: for example tower-http which is used by Axum supports gzip, br and deflate content encodings (see https://github.com/tower-rs/tower-http/pull/156). While deflate should be irrelevant, adding support for Brotli compression might be desirable in future.

Reply:
I’m perfectly fine with Gzip only. Brotli support is something for the future.